SciMovies Web Application

RileyManda opened the pull request:
**_Student Success have given me permission to continue this Capstone week solo. If you have any questions please reach out to Marco via the Code Reviewers Slack workspace._**

I have completed building the web application according to the stipulated  [requirements](https://github.com/RileyManda/javascript-capstone-project/issues/15):

The following implementations were made:

1. [Set up the project with Webpack and Jest- group task #18](https://github.com/RileyManda/scimovies-capstone-project/issues/18)

2.  Completed the task of [Display list of items on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/15) 

_**Implementations:**_

- [x] Created a config file to access API url.
- [x] Used the ID the data returned by the APi to display list of items to view
- [x] Implement a  card dom element to display tv show image,like button and tv show name
- [x] Implemented the comment button to the view
- [x] Used arrow functions.
- [x] Used async and await JavaScript features to consume the API.
- [x] Implemented error handling with try catch and snackbar on ui.


3. Completed the task of [Add all items counter on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/6) according to the stipulated  [requirements](https://github.com/RileyManda/javascript-capstone-project/issues/6):

_**Implementations:**_

- [x] Create feature: All items counter function and utils.
- [x] Displayed count of items on header.


- [x] Used [tvMazeAPI](https://www.tvmaze.com/api) to fetch data.

4. Completed the task of [Display number of likes for each item on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/12)
   

<img width="204" alt="Screen Shot 2023-07-06 at 5 04 43 PM" src="https://github.com/RileyManda/javascript-capstone-project/assets/37291194/dc9da4c6-7feb-4595-8e74-f60cb3298cbf">

  **Likes data is Posted to involvement API**

- [x] - Used POST to push likes to [Involvement API](https://microverse.notion.site/Involvement-API-869e60b5ad104603aa6db59e08150270)


<img width="400" alt="Screen Shot 2023-07-04 at 10 51 09 PM" src="https://github.com/RileyManda/javascript-capstone-project/assets/37291194/b639ef5d-c61b-4004-9341-9fc7fd2c0d83">




5. Completed the task of [ Add tests for items counter](https://github.com/RileyManda/scimovies-capstone-project/issues/3)


6. Completed the following group tasks:

[Add footer - group task #16](https://github.com/RileyManda/javascript-capstone-project/issues/16)
[Add header - group task #17](https://github.com/RileyManda/javascript-capstone-project/issues/17)

**And in addition:**
- Used arrow functions.
- Used async and await JavaScript features to consume the API.
- Implemented error handling
- Styled the UI using complimentary material theme colors.

I have also ensured that this project fulfills the following project requirements :

- [x] Used [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Implemented ES6 features(modules.
- [x] Used Fetch and promises
- [x] Used semantic HTML tags.
- [x] Used CSS selectors correctly.
- [x] Used JavaScript to manipulate DOM elements.
- [x] Used JavaScript events.
- [x] Used objects to store and access data.
- [x] There are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- [x] Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

- [x] Used JavaScript to make websites dynamic and build basic single page apps.
- [x] Used ES6 syntax.
- [x] Used ES6 modules.
- [x] Used callbacks and promises.
- [x] Used webpack.
- [x] Applied JavaScript best practices and language style guides in code.
- [x] Used AAA pattern for unit tests.
- [x] Wrote units tests 
- [x] Followed Gitflow.
- [x] Solved simple git conflicts.
- [x] Sent and received data from an Maze and Involvement API's.
- [x] Performed a [code review](https://github.com/RileyManda/javascript-capstone-project/pull/24) for a team member.

Please review my implementation and suggest any improvements and changes.

Thank you

**_Student Success have given me permission to continue this Capstone week solo. If you have any questions please reach out to Marco via the Code Reviewers Slack workspace._**

Zuhaib042 reviewed (changes requested):
Hi @RileyManda 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
![](https://media.giphy.com/media/m3864rBwwBTKMAbpn3/giphy.gif)

## Highlights
✅ Followed ES6 syntax
✅ Followed design given in the project description

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda commented:
**_Student Success have given me permission to continue this Capstone week solo. If you have any questions please reach out to Marco via the Code Reviewers Slack workspace._**

I have completed refactoring the project as requested by the previous reviewer:

**Iterations made:**

- [x] _Fixed the bug in add-likes component_
- [x] _Updated getTvShows,GetLikes functions to use async, await with fetch_
- [x] _Update readme file to fix broken link_

In addition, I have completed building the web application according to the stipulated  [requirements](https://github.com/RileyManda/javascript-capstone-project/issues/15):

The following implementations have been  made:

1. [Set up the project with Webpack and Jest- group task #18](https://github.com/RileyManda/scimovies-capstone-project/issues/18)

2.  Completed the task of [Display list of items on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/15) 

_**Implementations:**_

- [x] Created a config file to access API url.
- [x] Used the ID the data returned by the APi to display list of items to view
- [x] Implement a  card dom element to display tv show image,like button and tv show name
- [x] Implemented the comment button to the view
- [x] Used arrow functions.
- [x] Used async and await JavaScript features to consume the API.
- [x] Implemented error handling with try catch and snackbar on ui.


3. Completed the task of [Add all items counter on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/6) according to the stipulated  [requirements](https://github.com/RileyManda/javascript-capstone-project/issues/6):

_**Implementations:**_

- [x] Create feature: All items counter function and utils.
- [x] Displayed count of items on header.


- [x] Used [tvMazeAPI](https://www.tvmaze.com/api) to fetch data.

4. Completed the task of [Display number of likes for each item on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/12)
   

<img width="204" alt="Screen Shot 2023-07-06 at 5 04 43 PM" src="https://github.com/RileyManda/javascript-capstone-project/assets/37291194/dc9da4c6-7feb-4595-8e74-f60cb3298cbf">

  **Likes data is Posted to involvement API**

- [x] - Used POST to push likes to [Involvement API](https://microverse.notion.site/Involvement-API-869e60b5ad104603aa6db59e08150270)


<img width="400" alt="Screen Shot 2023-07-04 at 10 51 09 PM" src="https://github.com/RileyManda/javascript-capstone-project/assets/37291194/b639ef5d-c61b-4004-9341-9fc7fd2c0d83">




5. Completed the task of [ Add tests for items counter](https://github.com/RileyManda/scimovies-capstone-project/issues/3)


6. Completed the following group tasks:

[Add footer - group task #16](https://github.com/RileyManda/javascript-capstone-project/issues/16)
[Add header - group task #17](https://github.com/RileyManda/javascript-capstone-project/issues/17)

**And in addition:**
- Used arrow functions.
- Used async and await JavaScript features to consume the API.
- Implemented error handling
- Styled the UI using complimentary material theme colors.

I have also ensured that this project fulfills the following project requirements :

- [x] Used [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Implemented ES6 features(modules.
- [x] Used Fetch and promises
- [x] Used semantic HTML tags.
- [x] Used CSS selectors correctly.
- [x] Used JavaScript to manipulate DOM elements.
- [x] Used JavaScript events.
- [x] Used objects to store and access data.
- [x] There are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- [x] Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

- [x] Used JavaScript to make websites dynamic and build basic single page apps.
- [x] Used ES6 syntax.
- [x] Used ES6 modules.
- [x] Used callbacks and promises.
- [x] Used webpack.
- [x] Applied JavaScript best practices and language style guides in code.
- [x] Used AAA pattern for unit tests.
- [x] Wrote units tests 
- [x] Followed Gitflow.
- [x] Solved simple git conflicts.
- [x] Sent and received data from an Maze and Involvement API's.
- [x] Performed a [code review](https://github.com/RileyManda/javascript-capstone-project/pull/24) for a team member.

Please review my implementation and suggest any improvements and changes.

Thank you

**_Student Success have given me permission to continue this Capstone week solo. If you have any questions please reach out to Marco via the Code Reviewers Slack workspace._**

thecodechaser commented on src/view/domView.js:
```diff
@@ -0,0 +1,96 @@
+
+    // record likes END
+    card.appendChild(cardContent);
+
+    const commentButton = document.createElement('button');
+    commentButton.classList.add('comment-btn');
+    commentButton.textContent = 'Comments';
+    card.appendChild(commentButton);
+
+    tvshowList.appendChild(card);
+  });
+
```
- [ ] Great job fixing the issues requested by the previous reviewer. But I see a major issue, I'm not able to see the comment pop-up. Clicking on the comment button nothing happens and the comments pop-up doesn't open. Please make sure to fix this issue.

RileyManda replied:
Hi @thecodechaser 
I am working solo on this project.The comment popup is for Student B who left the team on Wednesday.
I added a disclaimer to my PR regrading this.
Please see the description.I will also attach it here:

Student Success have given me permission to continue this Capstone week solo. If you have any questions please reach out to Marco via the Code Reviewers Slack workspace.

thecodechaser reviewed (changes requested):
## STATUS: CHANGES REQUIRED ♻️ 

## Hi @RileyManda, and Team👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![image](https://github.com/RileyManda/scimovies-capstone-project/assets/83803414/5a6e87e2-7716-4f8a-b0ce-7657e4d62651)

## To Highlight 🎉
- All tests are passed. ✔️
- Nice UI design. ✔️
- No linter errors. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- Gitflow is followed. :heavy_check_mark:
- PR has a good title and summary. ✔️

<img src="https://media2.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif"/>

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

RileyManda commented:
### I have explained to the previous reviewer that I am working solo on this project. There is no Student B and C.

**_Student Success have given me permission to continue this Capstone week solo. If you have any questions please reach out to Marco via the Code Reviewers Slack workspace._**

I have completed refactoring the project as requested/recommended by the previous reviewer:(@Zuhaib042 )

**Iterations made:**

- [x] _Fixed the bug in add-likes component_
- [x] _Updated getTvShows,GetLikes functions to use async, await with fetch_
- [x] _Updated readme file to fix broken link_

In addition, I have completed building the web application according to the stipulated  [requirements](https://github.com/RileyManda/javascript-capstone-project/issues/15):

The following implementations have been  made:

1. [Set up the project with Webpack and Jest- group task #18](https://github.com/RileyManda/scimovies-capstone-project/issues/18)

2.  Completed the task of [Display list of items on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/15) 

_**Implementations:**_

- [x] Created a config file to access API url.
- [x] Used the ID the data returned by the APi to display list of items to view
- [x] Implement a  card dom element to display tv show image,like button and tv show name
- [x] Implemented the comment button to the view
- [x] Used arrow functions.
- [x] Used async and await JavaScript features to consume the API.
- [x] Implemented error handling with try catch and snackbar on ui.


3. Completed the task of [Add all items counter on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/6) according to the stipulated  [requirements](https://github.com/RileyManda/javascript-capstone-project/issues/6):

_**Implementations:**_

- [x] Create feature: All items counter function and utils.
- [x] Displayed count of items on header.


- [x] Used [tvMazeAPI](https://www.tvmaze.com/api) to fetch data.

4. Completed the task of [Display number of likes for each item on the Homepage](https://github.com/RileyManda/javascript-capstone-project/issues/12)
   

<img width="204" alt="Screen Shot 2023-07-06 at 5 04 43 PM" src="https://github.com/RileyManda/javascript-capstone-project/assets/37291194/dc9da4c6-7feb-4595-8e74-f60cb3298cbf">

  **Likes data is Posted to involvement API**

- [x] - Used POST to push likes to [Involvement API](https://microverse.notion.site/Involvement-API-869e60b5ad104603aa6db59e08150270)


<img width="400" alt="Screen Shot 2023-07-04 at 10 51 09 PM" src="https://github.com/RileyManda/javascript-capstone-project/assets/37291194/b639ef5d-c61b-4004-9341-9fc7fd2c0d83">




5. Completed the task of [ Add tests for items counter](https://github.com/RileyManda/scimovies-capstone-project/issues/3)


6. Completed the following group tasks:

[Add footer - group task #16](https://github.com/RileyManda/javascript-capstone-project/issues/16)
[Add header - group task #17](https://github.com/RileyManda/javascript-capstone-project/issues/17)

**And in addition:**
- Used arrow functions.
- Used async and await JavaScript features to consume the API.
- Implemented error handling
- Styled the UI using complimentary material theme colors.

I have also ensured that this project fulfills the following project requirements :

- [x] Used [Gitflow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/gitflow.md).
- [x] Implemented ES6 features(modules.
- [x] Used Fetch and promises
- [x] Used semantic HTML tags.
- [x] Used CSS selectors correctly.
- [x] Used JavaScript to manipulate DOM elements.
- [x] Used JavaScript events.
- [x] Used objects to store and access data.
- [x] There are [no linter errors](https://github.com/microverseinc/linters-config).
- [x] Used correct [GitHub flow](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/git-github/articles/github_flow.md).
- [x] Documented my work [in a professional way](https://github.com/microverseinc/curriculum-transversal-skills/blob/main/documentation/articles/professional_repo_rules.md).
- [x] Followed the list of [best practices for HTML & CSS](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/html_css_best_practices.md).
- [x] Followed the list of [best practices for JavaScript](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/javascript_best_practices.md).

- [x] Used JavaScript to make websites dynamic and build basic single page apps.
- [x] Used ES6 syntax.
- [x] Used ES6 modules.
- [x] Used callbacks and promises.
- [x] Used webpack.
- [x] Applied JavaScript best practices and language style guides in code.
- [x] Used AAA pattern for unit tests.
- [x] Wrote units tests 
- [x] Followed Gitflow.
- [x] Solved simple git conflicts.
- [x] Sent and received data from an Maze and Involvement API's.
- [x] Performed a [code review](https://github.com/RileyManda/javascript-capstone-project/pull/24) for a team member.

Please review my implementation and suggest any improvements and changes.

Thank you

**_Student Success have given me permission to continue this Capstone week solo. If you have any questions please reach out to Marco via the Code Reviewers Slack workspace._**

TadesseDev reviewed (approved):
### Hi @RileyManda 

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
![giphy (1)](https://user-images.githubusercontent.com/69077061/212969210-69f495ba-f2fb-4559-a222-d1507310d6fb.gif)



## Highlights
 - [x] Good job making the changes requested by the previous code reviewer. ✅
## Optional suggestions

- [Optional] Dear @RileyManda, It would be a great lesson to take the chance and implement parts of the project which are assigned to other students. This will teach you a lot and make your project complete. 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
